File: chat-tiempo-real-offline/frontend/chat-tiempo-real-offline/src/app/models/mensaje.model.ts
/*
  Modelo Mensaje.

  Representa un mensaje dentro del chat.

  estado:
  - enviado: el mensaje llegó correctamente al servidor.
  - pendiente: el mensaje se guardó localmente porque no había conexión.
*/

export type EstadoMensaje = 'enviado' | 'pendiente';

export interface Mensaje {
  id: string;
  usuarioId: number;
  usuarioNombre: string;
  texto: string;
  fecha: string;
  estado: EstadoMensaje;
}

File: chat-tiempo-real-offline/backend/models/mensaje_model.py
"""
mensaje_model.py

Modelo del mensaje del chat.

Debe coincidir con la estructura usada en Angular.
"""

from typing import Literal
from pydantic import BaseModel


class Mensaje(BaseModel):
    id: str
    usuarioId: int
    usuarioNombre: str
    texto: str
    fecha: str
    estado: Literal["enviado", "pendiente"]